### PR TITLE
Consolidate mirror logic and append 'mirror-' to page slugs

### PR DESCRIPTION
**Pull request by alfarok**
### Description
This PR addresses #3448 

It was determined that the duplicate route errors we were seeing were due to the implementation of `mirror.mjs`.  This file runs during builds to created duplicate pages to avoid some [restrictions of Docusaurus](https://github.com/w3f/polkadot-wiki/blob/master/scripts/mirror.mjs#L4).  The issue is that these duplicate mirrored pages share the same `slug` as the original pages.

The top of each markdown doc contains some metadata.  One of the properties in this metadata is the `slug`, which determines the route that will be used for that page.  `mirror.mjs` updates the `id` property only and keeps the route the same.  This is the source of the errors, 2 pages now share the same `slug`.

The router detects this, generating one set of errors and the redirect code throws again for the same reason (100+ lines of errors in console).  Since these pages share the same `slug`, whichever page is compiled later gets rendered for that route.  This isn't a big issue because the content is the same but it will always trigger the console errors.

### Current Solution
Append the same 'mirror-' prefix to the slug of the mirrored page.  All errors go away but mirror pages now have their own routes.  For example:

`docs/learn-crowdloans` and `docs/mirror-learn-crowdloans` would both be valid routes.

~~It's a bit strange to have both pages.  For example, how does this play with analytics.~~
^After looking into this more, I don't think this is a big issues.  The mirror pages are only ever used for Kusama content.  So for example, a Polkadot route in the docs could be 

`https://wiki.polkadot.network/docs/build-pdk`

while the Kusama route would be:

`https://guide.kusama.network/docs/mirror-build-pdk`

but

`https://wiki.polkadot.network/docs/mirror-build-pdk`

and 

`https://guide.kusama.network/docs/build-pdk`

would technically exist but never be referenced in the docs so a user would have to manually type out these type of routes.  For the purposes of analytics, it's very low likelihood that both routes would get regular hits and cause any confusion.  The `mirror-` prefix could be changed to `kusama-` for our purposes but then `https://wiki.polkadot.network/docs/kusama-build-pdk` would exists too, although never referenced.

### Alternative
Turn-off duplicate page errors in the `docusaurus.config.js` files via the `onDuplicateRoutes` property and still deal with the errors surfacing again at the [redirect code](https://github.com/w3f/polkadot-wiki/blob/master/polkadot-wiki/docusaurus.config.js#L84).  This risk here is hiding real errors.

**With either solution we should merge this PR as it consolidates `mirror.mjs`.  If we don't want the additional routes, the `else if` condition in the `BuildMirrors` function can be removed.**

### CODE REVIEW

**Review by emresurmeli — APPROVED** [review]

**Comment by alfarok** [comment]
I noticed the sidebar isn't showing on the routes with the `mirror-` prefix.  Looking into it now.  

@emresurmeli & @DrW3RK Do you guys have a preference on the current solution vs the alternative I outlined in the PR description?  There is a trade-off with either option.

**Comment by DrW3RK** [comment]
I have not run this on my machine yet. @emresurmeli What was the output at your end?

**Comment by DrW3RK** [comment]
I don't think I have the bandwidth to review this PR today, but to give you a backdrop of where this all started https://github.com/w3f/polkadot-wiki/issues/411 

Also, search for "mirror" in the repo's issues and PRs and you'll find the work that has been put into this

**Comment by alfarok** [comment]
No rush at all, I just want to discuss it a little before merging since it can be handled a couple different ways with different repercussions.

I know some of these infrastructure related PRs aren't very exciting but we almost have all the errors/warnings from the build process handled.  It should make it a lot easier to catch new bugs and issues earlier moving forward.

**Comment by alfarok** [comment]
I spent a little more time trying to understand how the mirrored pages are currently used.  I now think we can remove the mirror functionality completely.  I should have tried this before attempting to salvage the existing system.  I realized some changes between Docusaurus v1 and v2 that I think eliminated the need for this work around.  The history you pointed me to @DrW3RK shows this was implemented when we were still using v1.

TLDR: removing all the mirror code and removing any `mirror-` prefixes from existing links doesn't seem to be an issue or break anything.  Despite the cleanup effort here, I am thinking about closing this PR and opening another that removes mirrors and all references to mirrors.

Let me know what you guys think.  In general, I think the more legacy code we can remove the better.  The more lean the repo is, the easier it is for a new potential contributor to get familiar with the code.

**Comment by DrW3RK** [comment]
Yeah, sounds good to me!
